test/rollup.js

```diff
@@ -19,10 +19,11 @@ test('xdm (rollup)', async function (t) {
     external: ['react/jsx-runtime'],
     plugins: [
       {
+        name: 'xdm',
         async transform(contents, filePath) {
           if (path.extname(filePath) !== '.mdx') return null
           var file = await compile({contents, path: filePath})
-          return {code: file.contents, map: file.map}
+          return {code: String(file.contents), map: file.map?.toString()}
```

Comment:
`toString` is incorrect: `file.map` is an (optional) JSON object.

Also: passing the JSON should be fine: https://github.com/rollup/rollup/blob/850d30a463b0b50f409413081629e33c81139300/src/rollup/types.d.ts#L101

(btw: rollup can apparently also get an acorn AST! that might be a nice performance optimization! I looked into passing ASTs for webpack’s babel loader but it doesn’t support it apparently)